libs/@local/hql/README.md

```diff
@@ -0,0 +1,51 @@
+# HQL - HASH Query Language
+
+This is the compiler, planner and executor for the HASH Query Language (HQL). HQL is a functional programming language that is created using a subset of JSON.
+
+## Language
+
+Language design documentation and a specification of the language will be provided in the future.
+
+The CST of the language is very basic, only consisting of three different constructs:
+
+- Function Invocation: A function invocation has multiple possible forms:
+  - `[name, ...args]`
+  - `{"fn": name, "args": args}`
```

Comment:
Why do we have two forms for calling a function? Isn't one enough?

Reply:
in case of ambiguity.

I wanted to have an explicit version - the object representation and a more "comfortable" representation (the list representation.

Once parsed, they are the same, but this helps us in multiple scenarios:

* cases of ambiguity, this was especially prevalent when `do` was it's own construct, not just a special form function
* machine tools. We're trying to bridge the gap a bit between easy machine representation and a representation that can easily be authored by a human, which reduces nesting.

So to accomodate all these together, we have the S-expr like for `[name, ...args]`, and the object like form `{"fn": name, "args": args}`.

Reply:
The CST represents the actual syntax, I really don't see why we would need two representation for one and the same syntax. Also, I don't like encoding function calls as arrays btw ;)